processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -385,19 +385,20 @@ private void initLongData()
           if (hasLowerBound()) {
             final Long lowerLong = GuavaUtils.tryParseLong(lower);
             if (lowerLong == null) {
-              matchesAnything = false;
-              return;
+              // Unparseable values fall before all actual numbers, so all numbers will match the lower bound.
+              hasLowerLongBoundVolatile = false;
+            } else {
+              hasLowerLongBoundVolatile = true;
+              lowerLongBoundVolatile = lowerLong;
             }
-
-            hasLowerLongBoundVolatile = true;
-            lowerLongBoundVolatile = lowerLong;
           } else {
             hasLowerLongBoundVolatile = false;
```

Comment:
@leventov I agree with you technically and the HB set by the write and read of `longsInitialized` . Now this is a small anonymous class where it is easy to see whole class and reason about HB relationship, that is why I said it makes small difference here. But , note that @gianm had to put a comment there to explain the HB relationship.

Now, think about a larger class and people relying too much on the HB set by one volatile variable to make sure other variable reads will happen after their writes... all is well. but , later someone modifies the code in a way that breaks the original HB assumption without remembering to now update other variables to be volatile and suddenly we would have a bug.